5calls/app/src/main/java/org/a5calls/android/a5calls/MainActivity.java

```diff
@@ -248,68 +244,105 @@ public void updateIssue(Issue issue) {
+                String code = pref.getString(KEY_USER_ZIP, "");
+                if (!TextUtils.isEmpty(code)) {
+                    onZipUpdated(code);
+                    holder.updateZipUi(false, Integer.parseInt(code));
                 }
-            });
-            int totalCalls = issue.contacts.length;
-            List<String> contacted = AppSingleton.getInstance(getApplicationContext())
-                    .getDatabaseHelper().getCallsForIssueAndZip(issue.id, mZip);
-            int callsLeft = totalCalls - contacted.size();
-            if (callsLeft == totalCalls) {
-                if (totalCalls == 1) {
-                    holder.numCalls.setText(getResources().getString(R.string.call_count_one));
+
+                // TODO: Option to get user's location from GPS instead of just entering a zip code.
+                holder.mZipCodeEditText.setOnEditorActionListener(new TextView.OnEditorActionListener() {
+                    @Override
+                    public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                        if (actionId == EditorInfo.IME_ACTION_DONE) {
+                            submitZip(holder.mZipCodeEditText.getText().toString(), holder.mZipCodeEditText);
+                            return true;
+                        }
+                        return false;
+                    }
+                });
+                holder.mZipCodeSubmitButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        submitZip(holder.mZipCodeEditText.getText().toString(), holder.mZipCodeEditText);
+                    }
+                });
+                holder.mZipCodeEditButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        holder.updateZipUi(true, 0);
+                    }
+                });
+            } else {
+                final Issue issue = mIssues.get(position - 1);
+
+                final IssueViewHolder holder = (IssueViewHolder) viewHolder;
+
+                holder.name.setText(issue.name);
+                holder.itemView.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        Intent issueIntent = new Intent(holder.itemView.getContext(),
+                                IssueActivity.class);
+                        issueIntent.putExtra(IssueActivity.KEY_ISSUE, issue);
+                        issueIntent.putExtra(IssueActivity.KEY_ZIP, mZip);
+                        startActivityForResult(issueIntent, ISSUE_DETAIL_REQUEST);
+                    }
+                });
+                int totalCalls = issue.contacts.length;
+                List<String> contacted = AppSingleton.getInstance(getApplicationContext())
```

Comment:
This puts a lot of the logic / database work in the Adapter. I'd prefer a simpler adapter with more complexity in the activity, or another handler class.